### PR TITLE
Added support for fast-m1

### PR DESCRIPTION
**Pull request by kaletap**

### CODE REVIEW

**Review by ardcore — CHANGES_REQUESTED** [review]
Suggestion: instead of a boolean `use_fast_m1` parameter, we could:
1. create an enum called `SearchEngine`, with variants `FAST_M1` and `CLASSIC` (there may be a better name :P)
2. have a parameter called `search_engine` accepting this enum, defaulting to `SearchEngine.CLASSIC`

This way we can easily add more "engines" in the future without introducing breaking changes in the API. It's also more coherent with other parameters, such as `detail_level` or `priority`.

@kaletap What do you think about this?

**Comment by kaletap** [comment]
That's a very good point, thanks. I fixed it now. I used standard class instead of enum, because I think that it would be better if user could pass string directly rather than import `SearchEngine` and provide `SearchEngine.CLASSIC` or `SearchEngine.FAST_M1`. What do you think?

**Review by ardcore — CHANGES_REQUESTED** [review]
Thanks @kaletap!
I agree. In python 3.11, we'll be able to use StrEnum, but python 3.11 is not released as stable yet.

File: m1wrapper/search.py [hunk 1]
```diff
@@ -38,11 +45,13 @@ def __init__(
         starting_materials=None,
         name=None,
         targets_metadata=None,
+        search_engine=False
```
[the remark this answers is not in the document]
Comment (kaletap) [reply m1wrapper/search.py]: No, fixed it now, thanks.

**Comment by ardcore** [comment]
Thanks @kaletap. The code looks good. @brunokam can you please test this manually and merge if everything works? Thanks!

**Review by ardcore — APPROVED** [review]

**Review by ardcore — APPROVED** [review]
LGTM :)

**Pull request closed without merging**